components/feature/session/src/main/java/mozilla/components/feature/session/SessionUseCases.kt

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session
+
+import android.net.Uri
+import android.text.TextUtils
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.Engine
+
+/**
+ * Contains all use cases related to the session feature. Every public function represents a single
+ * use case.
+ */
+class SessionUseCases(
+    private val sessionManager: SessionManager,
+    private val engine: Engine,
+    private val sessionMapping: SessionMapping
+) {
+
+    /**
+     * Loads the provided URL for the currently selected session.
+     *
+     * @param url url to load.
+     */
+    val loadUrl = fun (url: String) {
+        val trimmedInput = url.trim({ it <= ' ' })
+        var uri = Uri.parse(trimmedInput)
+        if (TextUtils.isEmpty(uri.scheme)) {
+            uri = Uri.parse("http://" + trimmedInput)
+        }
```

Comment:
I wonder what parts the interactor should handle and what is the usecase's job. Where's the boundary? E.g. Would input sanitization be part of the use case or interactor?

Reply:
Good question. I think it depends. If the sanitization is specific to the UI, then it should be the UI and interactor, but if we want it to be re-usable it should be done in the use-case.